specs/AppLifecycle/Activation/AppLifecycle Activation.md

```diff
@@ -0,0 +1,248 @@
+
+|     Activation   kind    |     Description                                                                                                                                             |
+|--------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------|
+|     Launch               |     Launch   from a Start tile/shortcut, from a command window, or programmatically via   ShellExecute/CreateProcess.                                       |
+|     File                 |     The   app is activated when the user double-clicks a registered file, or when a   caller calls ShellExecute or LaunchFileAsync on a registered file.    |
+|     Protocol             |     Activated   via ShellExecute or LaunchUriAsync, or by specifying a protocol string on a   command-line.                                                 |
+|     StartupTask          |     The   app starts on user log-in - either registered in the registry, or via   shortcuts in a well-known startup folder.                                 |
+|     ToastNotification    |     The   app sends a toast, the user responds, and this activates the app.                                                                                 |
+    
+
+### Activation registration
+From a Win32 app perspective, the 2 activation kinds that do not require any registration (Shortcut/Tile Launch and CommandLine) are effectively the same in terms of activation payload. Indeed, it is not possible to distinguish a launch activation of a Win32 app from a commandline activation. Therefore, the AppLifecycle implementation will merge Launch and CommandLine into one (Launch) activation kind. 
+
+UWP apps can only register for activation kinds via manifest declarations, and therefore this happens during app install. Conversely, traditional app types can write registry entries at any time.
+
+A key requirement is to enable unpackaged Win32 apps to use the set of rich activations using familiar designs, without any additional friction. Specifically, AppLifecycle will provide APIs which the app can call to register its interest in rich activations. The app is not required to have an MSIX manifest, nor an MSIX package. Using the API approach rather than the manifest approach has the added benefit that it also addresses a related customer request for dynamic activation registrations even for UWP apps.
+
+Ideally, if the app is installed per-machine then any activation registrations would also be per-machine; and if installed per-user then registrations would be per-user. The platform will not offer any additional API features for this - that is, the app doesn't get to choose whether its activation registrations are per-machine or per-user. Instead, the activations should be registered to match the way the app is installed. For v1, the platform will support only per-user registration.
```

Comment:
This is a little confusing. Maybe lead off with "Currently all registered activation points are managed per-user. If your app is installed for multiple users via an MSI or elevated installer your app will need to re-register launch points for every user. Consider using {this sample technique} to ensure registrations only happen once per user."

Reply:
Updated.